### PR TITLE
Auth device list component

### LINKED ISSUE #36230: UI components for account settings sections

### PR DESCRIPTION
**Pull request by bl-nero**
Introduces new device list components that will be used on the new account settings screen.

[Figma design](https://www.figma.com/file/8DQZl1mYUPLgvKgmA6zPOM/In-Progress-Work?node-id=7085%3A28403&mode=dev)

Screenshot from the story:
Screenshot 2024-01-03 at 17 06 46

Closes #36230

### CODE REVIEW

File: web/packages/teleport/src/Account/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      <Box
+        bg={theme.colors.interactive.tonal.neutral[0]}
+        lineHeight={0}
+        p={2}
+        borderRadius={3}
+      >
+        {icon}
+      </Box>
+      <Box flex="1">
+        <Text typography="h4">{title}</Text>
```
Comment (zmb3) [inline web/packages/teleport/src/Account/Header.tsx:27]: Why `h4`?

File: web/packages/teleport/src/Account/ManageDevices/AuthDeviceList/AuthDeviceList.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  return (
+    <Cell>
+      <ButtonWarningBorder p={2} onClick={onRemove}>
+        <Icon.Trash size="small" />
+      </ButtonWarningBorder>
+    </Cell>
+  );
+}
+
+const StyledTable = styled(Table)(
```
Comment (zmb3) [inline web/packages/teleport/src/Account/ManageDevices/AuthDeviceList/AuthDeviceList.tsx:88]: Comparing the design and the storybook - I can't quite put my finger on it but Figma looks a bit "crisper" - could it be the font weights?

**Review by zmb3 — APPROVED** [review]

File: web/packages/teleport/src/Account/Header.tsx (hunk 1 above)
Comment (bl-nero) [reply web/packages/teleport/src/Account/Header.tsx:27]: It was the closest match that I found to the designs. And this being a header, seems to fit the purpose. What's your suggestion?

**Review by rudream — APPROVED** [review]

File: web/packages/teleport/src/Account/Header.tsx (hunk 1 above)
Comment (bl-nero) [reply web/packages/teleport/src/Account/Header.tsx:27]: (Actually, to clarify: the design, if you inspect the typography info, calls for "Ubuntu/H2" typography, but the font size corresponds to our h4, so I decided to use that one.)

Comment (zmb3) [reply web/packages/teleport/src/Account/Header.tsx:45]: In the future I suggest we don't choose an element based on what font size we get by default, and instead choose the most semantically correct element and use CSS to get the desired font size.

Having an h4 that doesn't fall under an h3 is semantically incorrect and not good for accessibility.

**Pull request merged**